Chore/update oss api (#3341)
influxdata/docs-v2: api-docs/README.md

InfluxDB uses [Redoc](https://github.com/Redocly/redoc/),
[redoc-cli](https://github.com/Redocly/redoc/blob/master/cli/README.md),
and Redocly's [OpenApi CLI](https://redoc.ly/docs/cli/) to generate

> API documentation from the InfluxDB openapi contracts.

Author: ```suggestion
API documentation from the InfluxDB openapi contracts (`contracts/ref`).
```

The structure versions swagger files using the following pattern:

```
api-docs/
  ├── v2.0/
  │     └── ref.yml
  ├── v2.1/
  │     └── ref.yml
  ├── v2.2/
  │     └── ref.yml
  └── etc...
```

### Configure OpenAPI CLI linting and bundling

> `.redoc.yaml` sets linting and bundling options for `openapi` CLI.

Author: ```suggestion
`.redoc.yaml` sets linting and bundling options for the `openapi` CLI.
```

`./openapi/plugins` contains custom OpenAPI CLI plugins composed of *rules* (for linting) and *decorators* (for bundle customization).

### Custom content

> `./openapi/content` contains custom OAS (OpenAPI Spec) content in YAML files. The content structure and Markdown must be valid OAS.

Author: ```suggestion
`./openapi/content` contains custom [OAS (OpenAPI Spec)](https://swagger.io/specification/) content in YAML files. The content structure and Markdown must be valid OAS.
```

`./openapi/plugins` use `./openapi/plugins/decorators` to apply the content to the contracts.
`.yml` files in `./openapi/content/` set content for sections (nodes) in the contract. To update the content for those nodes, you only need to update the YAML files.

> To add new YAML files for other nodes in the openapi contracts, configure the new content YAML file in `./openapi/content/content.js`. Then, write a decorator module for the node and configure the decorator in the plugin, e.g. `./openapi/plugins/docs-plugin.js`. See the [complete list of OAS v3.0 nodes](https://github.com/Redocly/openapi-cli/blob/master/packages/core/src/types/oas3.ts#L529).

Author: ```suggestion
To add a new section (node) to the openapi contracts, add a YAML file for the node in `./openapi/content/content.js`. Then, write a decorator module for the node and configure the decorator in the plugin, e.g. `./openapi/plugins/docs-plugin.js`. See the [complete list of OAS v3.0 nodes](https://github.com/Redocly/openapi-cli/blob/master/packages/core/src/types/oas3.ts#L529).
```
